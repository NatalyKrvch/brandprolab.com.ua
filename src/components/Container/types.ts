export interface ContainerProps {
  fullWidth?: boolean;
}
