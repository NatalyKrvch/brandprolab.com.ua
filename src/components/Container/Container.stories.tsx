import type { Meta, StoryObj } from '@storybook/react';
import Container from './Container';

const meta: Meta<typeof Container> = {
  title: 'Components/Container',
  component: Container,
  parameters: {
    controls: { expanded: true },
  },
  argTypes: {
    fullWidth: {
      control: { type: 'boolean' },
      description:
        'If true, the container will span the full width of the screen.',
    },
    children: {
      control: { type: 'text' },
      description: 'The content to be rendered inside the container.',
    },
  },
};

export default meta;
type Story = StoryObj<typeof Container>;

export const Default: Story = {
  args: {
    fullWidth: false,
    children: (
      <div className="text-base">
        This is a default container with max-width 1224px.
      </div>
    ),
  },
};

export const FullWidth: Story = {
  args: {
    fullWidth: true,
    children: <div className="text-base">This is a full-width container.</div>,
  },
};
