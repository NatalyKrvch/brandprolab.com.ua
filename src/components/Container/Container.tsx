import {
  FULL_WIDTH_CONTAINER_TEST_ID,
  LIMITED_WIDTH_CONTAINER_TEST_ID,
} from '@/lib/constants';
import { ContainerProps } from './types';

const Container = ({ fullWidth, children }: ContainerProps) => {
  if (fullWidth) {
    return (
      <div className="mx-auto" data-testid={FULL_WIDTH_CONTAINER_TEST_ID}>
        {children}
      </div>
    );
  }

  return (
    <div
      className="mx-auto max-w-[1224px] px-4 tablet:px-12 desktop:px-12"
      data-testid={LIMITED_WIDTH_CONTAINER_TEST_ID}
    >
      {children}
    </div>
  );
};

export default Container;
