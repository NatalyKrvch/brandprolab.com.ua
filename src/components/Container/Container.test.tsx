import { render, screen } from '@testing-library/react';
import Container from './Container';
import '@testing-library/jest-dom';
import {
  FULL_WIDTH_CONTAINER_TEST_ID,
  LIMITED_WIDTH_CONTAINER_TEST_ID,
} from '@/lib/constants';

describe('Container component', () => {
  it('renders children inside the full-width container', () => {
    const { asFragment } = render(
      <Container fullWidth>
        <p>Test content</p>
      </Container>,
    );

    const container = screen.getByTestId(FULL_WIDTH_CONTAINER_TEST_ID);
    expect(container).toBeInTheDocument();
    expect(container).toHaveTextContent('Test content');

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders children inside the limited-width container', () => {
    const { asFragment } = render(
      <Container>
        <p>Default content</p>
      </Container>,
    );

    const container = screen.getByTestId(LIMITED_WIDTH_CONTAINER_TEST_ID);
    expect(container).toBeInTheDocument();
    expect(container).toHaveTextContent('Default content');

    expect(asFragment()).toMatchSnapshot();
  });

  it('applies fullWidth styles when fullWidth is true', () => {
    const { asFragment } = render(
      <Container fullWidth>
        <p>Full width content</p>
      </Container>,
    );

    const container = screen.getByTestId(FULL_WIDTH_CONTAINER_TEST_ID);
    expect(container).toHaveClass('mx-auto');

    expect(asFragment()).toMatchSnapshot();
  });

  it('applies default styles when fullWidth is false', () => {
    const { asFragment } = render(
      <Container>
        <p>Default content</p>
      </Container>,
    );

    const container = screen.getByTestId(LIMITED_WIDTH_CONTAINER_TEST_ID);
    expect(container).toHaveClass('mx-auto');
    expect(container).toHaveClass('max-w-[1224px]');
    expect(container).toHaveClass('px-4');

    expect(asFragment()).toMatchSnapshot();
  });

  it('applies tablet and desktop styles correctly', () => {
    const { asFragment } = render(
      <Container>
        <p>Responsive content</p>
      </Container>,
    );

    const container = screen.getByTestId(LIMITED_WIDTH_CONTAINER_TEST_ID);
    expect(container).toHaveClass('tablet:px-12');
    expect(container).toHaveClass('desktop:px-12');

    expect(asFragment()).toMatchSnapshot();
  });
});
