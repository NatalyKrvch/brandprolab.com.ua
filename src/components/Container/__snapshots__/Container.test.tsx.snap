// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component applies default styles when fullWidth is false 1`] = `
<DocumentFragment>
  <div
    class="mx-auto max-w-container px-4 tablet:px-12 desktop:px-12"
    data-testid="limited-width-container-test-id"
  >
    <p>
      Default content
    </p>
  </div>
</DocumentFragment>
`;

exports[`Container component applies fullWidth styles when fullWidth is true 1`] = `
<DocumentFragment>
  <div
    class="mx-auto"
    data-testid="full-width-container-test-id"
  >
    <p>
      Full width content
    </p>
  </div>
</DocumentFragment>
`;

exports[`Container component applies tablet and desktop styles correctly 1`] = `
<DocumentFragment>
  <div
    class="mx-auto max-w-container px-4 tablet:px-12 desktop:px-12"
    data-testid="limited-width-container-test-id"
  >
    <p>
      Responsive content
    </p>
  </div>
</DocumentFragment>
`;

exports[`Container component renders children inside the full-width container 1`] = `
<DocumentFragment>
  <div
    class="mx-auto"
    data-testid="full-width-container-test-id"
  >
    <p>
      Test content
    </p>
  </div>
</DocumentFragment>
`;

exports[`Container component renders children inside the limited-width container 1`] = `
<DocumentFragment>
  <div
    class="mx-auto max-w-container px-4 tablet:px-12 desktop:px-12"
    data-testid="limited-width-container-test-id"
  >
    <p>
      Default content
    </p>
  </div>
</DocumentFragment>
`;
